### PR TITLE
 logs: k8s-file: restore poll sleep

### PR DESCRIPTION
**Pull request by vrothberg**
Commit 84b55ee attempted to fix a race waiting for the container
died event.  Previously, Podman slept for duration of the polling
frequence which I considerred to be a mistake.  As it turns out, I was
mistaken since the file logger will, in fact, NOT read until EOF and
then stop logging but stop logging immediately _after_ it woke up.

[NO TESTS NEEDED] as the race condition cannot be hit reliably.

Fixes: #10675
Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
Needs a backport to v3.2.

@containers/podman-maintainers PTAL

File: go.mod [hunk 1]
```diff
@@ -42,6 +42,7 @@ require (
 	github.com/mattn/go-colorable v0.1.8 // indirect
 	github.com/moby/term v0.0.0-20201216013528-df9cb8a40635
 	github.com/mrunalp/fileutils v0.5.0
+	github.com/nxadm/tail v1.4.8
```
Comment (Luap99) [inline go.mod:45]: This looks wrong, podman uses `github.com/hpcloud/tail`

File: libpod/container_log.go [hunk 2]
```diff
@@ -4,10 +4,12 @@ import (
 	"context"
 	"fmt"
 	"os"
+	"time"
 
 	"github.com/containers/podman/v3/libpod/define"
 	"github.com/containers/podman/v3/libpod/events"
 	"github.com/containers/podman/v3/libpod/logs"
+	"github.com/nxadm/tail/watch"
```
Comment (Luap99) [inline libpod/container_log.go:12]: same here

Comment (vrothberg) [reply libpod/container_log.go:12]: Good catch!  Looks like my vim plugin chose the wrong dep.  Thanks!

File: libpod/container_log.go [hunk 3]
```diff
@@ -93,17 +95,20 @@ func (c *Container) readFromLogFile(ctx context.Context, options *logs.LogOption
...
+		// getting its state, instruct the logger to read the file
+		// until EOF.
 		state, err := c.State()
 		if err != nil || state != define.ContainerStateRunning {
-			// If the container isn't running or if we encountered
-			// an error getting its state, instruct the logger to
-			// read the file until EOF.
+			// Make sure to wait at least for the poll duration
+			// before stopping the file logger (see #10675).
+			time.Sleep(watch.POLL_DURATION)
```
Comment (TomSweeneyRedHat) [inline libpod/container_log.go:105]: I'm always worried about putting a hard sleep into a function like this, but I'm not thinking of a better way atm.

Comment (vrothberg) [reply libpod/container_log.go:105]: Same here. I _really_ did not like it which is why I removed it previously.  I read the docs of the library we're using but they did not reflect reality :(

**Review by Luap99 — COMMENTED** [review]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**